Python/Product/DebuggerHelper/trace.cpp

```diff
@@ -524,21 +524,19 @@ static void TraceReturn(void* frame) {
 #pragma warning(disable:4211) // nonstandard extension used: redefined extern to static
 
 static void ReleasePendingObjects() {
-    if (objectsToRelease) {
-        auto otr = reinterpret_cast<ObjectToRelease*>(InterlockedExchange64(&objectsToRelease, 0));
-        while (otr != nullptr) {
-            // Releasing an object may trigger execution of its __del__ function, which will cause re-entry to this code,
-            // so null out the reference before releasing, and check for nulls in the list.
-            auto obj = otr->pyObject;
-            if (obj != 0) {
-                otr->pyObject = 0;
-                Py_DecRef(reinterpret_cast<PyObject*>(obj));
-            }
-
-            auto next = reinterpret_cast<ObjectToRelease*>(otr->next);
-            VirtualFree(otr, 0, MEM_RELEASE);
-            otr = next;
+    auto otr = reinterpret_cast<ObjectToRelease*>(InterlockedExchange64(&objectsToRelease, 0));
```

Comment:
![image](https://user-images.githubusercontent.com/3315992/53908663-b7d8db00-4004-11e9-82e6-9285f90662f9.png)

Reply:
This is a false positive, because `objectsToRelease` are only ever going to be set from the debugger while the debuggee (and this code inside it) is frozen - and it's marked `volatile` to indicate that such external modification can take place. The use of `InterlockedExchange` here is to ensure that the swap is atomic even in the face of debugger pausing the process - if you look at the code, there's no other interlocked operation on this variable.

The change will work, but this function is called on a hot path (from `TraceFunc`, which invokes it on every call - i.e. this is called for every line of Python code traced), so avoiding an atomic operation if we can do the check without is beneficial to perf.

Reply:
Makes sense. Let's add that explanation as a comment, and suppress that warning locally.

Reply:
Will be updated in next commit